On OSX - default BSD grep breaks gen_conversion

Using the grep from OSX, fails when GREP_OPTIONS/GREP_COLOR is set as shown below (`GREP_OPTIONS=--color=always`).

Workaround : unset GREP_OPTIONS/GREP_COLOR

Note that using GNU grep using brew install will show that GREP_OPTIONS shows immediately that GREP_OPTIONS may be interfering with make (http://paste.openstack.org/show/533072/)

```
[dims@dims-mac 12:23] ~/go/src/k8s.io/kubernetes ⟩ make gen_conversion
Makefile.generated_files:265: warning: overriding commands for target `_output/bin/deepcopy-gen'
.make/_output/bin/deepcopy-gen.mk:2: warning: ignoring old commands for target `_output/bin/deepcopy-gen'
Makefile.generated_files:441: warning: overriding commands for target `_output/bin/conversion-gen'
.make/_output/bin/conversion-gen.mk:2: warning: ignoring old commands for target `_output/bin/conversion-gen'
Go version: go version go1.6.2 darwin/amd64
+++ [0715 12:24:08] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [0715 12:24:09] Building go targets for darwin/amd64:
    cmd/libs/go2idl/conversion-gen
+++ [0715 12:24:11] Placing binaries
I0715 12:24:14.089090   23072 execute.go:214] Processing package "v1alpha1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/apps/v1alpha1"
E0715 12:24:14.091458   23072 conversion.go:592] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/apps/v1alpha1.PetSetSpec <-> k8s.io/kubernetes/pkg/apis/apps.PetSetSpec
I0715 12:24:14.092023   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/apps/v1alpha1/zz_generated.conversion.go"
I0715 12:24:14.094486   23072 execute.go:214] Processing package "v1beta1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/authentication.k8s.io/v1beta1"
I0715 12:24:14.097154   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/authentication.k8s.io/v1beta1/zz_generated.conversion.go"
I0715 12:24:14.098660   23072 execute.go:214] Processing package "v1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/batch/v1"
I0715 12:24:14.105808   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/batch/v1/zz_generated.conversion.go"
I0715 12:24:14.108424   23072 execute.go:214] Processing package "v1alpha1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/certificates/v1alpha1"
I0715 12:24:14.112448   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/certificates/v1alpha1/zz_generated.conversion.go"
I0715 12:24:14.116189   23072 execute.go:214] Processing package "v1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/autoscaling/v1"
I0715 12:24:14.123928   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/autoscaling/v1/zz_generated.conversion.go"
I0715 12:24:14.126449   23072 execute.go:214] Processing package "v1beta1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/authorization/v1beta1"
I0715 12:24:14.132244   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/authorization/v1beta1/zz_generated.conversion.go"
I0715 12:24:14.134912   23072 execute.go:214] Processing package "v1alpha1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/rbac/v1alpha1"
I0715 12:24:14.144705   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/rbac/v1alpha1/zz_generated.conversion.go"
I0715 12:24:14.151205   23072 execute.go:214] Processing package "v2alpha1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/batch/v2alpha1"
I0715 12:24:14.162494   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/batch/v2alpha1/zz_generated.conversion.go"
I0715 12:24:14.166776   23072 execute.go:214] Processing package "v1alpha1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/policy/v1alpha1"
I0715 12:24:14.171686   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/policy/v1alpha1/zz_generated.conversion.go"
I0715 12:24:14.173432   23072 execute.go:214] Processing package "v1beta1", disk location "/Users/dims/go/src/k8s.io/kubernetes/federation/apis/federation/v1beta1"
I0715 12:24:14.178143   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/federation/apis/federation/v1beta1/zz_generated.conversion.go"
I0715 12:24:14.181795   23072 execute.go:214] Processing package "v1beta1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
E0715 12:24:14.202574   23072 conversion.go:592] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.HorizontalPodAutoscalerSpec <-> k8s.io/kubernetes/pkg/apis/autoscaling.HorizontalPodAutoscalerSpec
E0715 12:24:14.211762   23072 conversion.go:592] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.JobSpec <-> k8s.io/kubernetes/pkg/apis/batch.JobSpec
E0715 12:24:14.231489   23072 conversion.go:592] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.RollingUpdateDeployment <-> k8s.io/kubernetes/pkg/apis/extensions.RollingUpdateDeployment
E0715 12:24:14.233595   23072 conversion.go:592] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.ScaleStatus <-> k8s.io/kubernetes/pkg/apis/extensions.ScaleStatus
I0715 12:24:14.240747   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/extensions/v1beta1/zz_generated.conversion.go"
I0715 12:24:14.265071   23072 execute.go:214] Processing package "v1alpha1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1"
I0715 12:24:14.274797   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1/zz_generated.conversion.go"
I0715 12:24:14.278519   23072 execute.go:214] Processing package "v1", disk location "/Users/dims/go/src/k8s.io/kubernetes/pkg/api/v1"
I0715 12:24:14.402447   23072 execute.go:67] Assembling file "/Users/dims/go/src/k8s.io/kubernetes/pkg/api/v1/zz_generated.conversion.go"
I0715 12:24:14.453891   23072 main.go:73] Completed successfully.
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩
[dims@dims-mac 12:25] ~/go/src/k8s.io/kubernetes ⟩ grep --version
grep (BSD grep) 2.5.1-FreeBSD
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩ export GREP_OPTIONS='--color=always'
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩ export GREP_COLOR='1;35;40'
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩ find . -name zz_generated.conversion.go -delete
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩
[dims@dims-mac 12:24] ~/go/src/k8s.io/kubernetes ⟩
[dims@dims-mac 12:25] ~/go/src/k8s.io/kubernetes ⟩ make gen_conversion
Makefile.generated_files:265: warning: overriding commands for target `_output/bin/deepcopy-gen'
.make/_output/bin/deepcopy-gen.mk:2: warning: ignoring old commands for target `_output/bin/deepcopy-gen'
Makefile.generated_files:441: warning: overriding commands for target `_output/bin/conversion-gen'
.make/_output/bin/conversion-gen.mk:2: warning: ignoring old commands for target `_output/bin/conversion-gen'
Makefile.generated_files:265: warning: overriding commands for target `_output/bin/deepcopy-gen'
.make/_output/bin/deepcopy-gen.mk:2: warning: ignoring old commands for target `_output/bin/deepcopy-gen'
Makefile.generated_files:441: warning: overriding commands for target `_output/bin/conversion-gen'
.make/_output/bin/conversion-gen.mk:2: warning: ignoring old commands for target `_output/bin/conversion-gen'
make: *** No rule to make target `.make/federation/apis/federation/gofiles.mk.stamp', needed by `federation/apis/federation/v1beta1/zz_generated.conversion.go'.  Stop.

```

--- a/Makefile.generated_files
+++ b/Makefile.generated_files
@@ -140,7 +140,7 @@ $(foreach dir, $(ALL_GO_DIRS), $(eval           \
 # but we only need to rebuild targets if the contents actually changed.  That
 # is what the .stamp file represents.
 $(foreach dir, $(ALL_GO_DIRS), $(META_DIR)/$(dir)/$(GOFILES_META)):
-	FILES=$$(ls $</*.go | grep -v $(GENERATED_FILE_PREFIX));  \
+	FILES=$$(ls $</*.go | grep --color=never -v $(GENERATED_FILE_PREFIX));  \
 	mkdir -p $(@D);                                           \
 	echo "gofiles__$< := $$(echo $${FILES})" >$@.tmp;         \
 	cmp -s $@.tmp $@ || touch $@.stamp;                       \
@@ -161,7 +161,7 @@ ifeq ($(DBG_MAKEFILE),1)
 endif
 ALL_K8S_TAG_FILES := $(shell                             \
     find $(ALL_GO_DIRS) -maxdepth 1 -type f -name \*.go  \
-        | xargs grep -l '^// *+k8s:'                     \
+        | xargs grep --color=never -l '^// *+k8s:'       \
 )
 
 #
@@ -188,7 +188,7 @@ ifeq ($(DBG_MAKEFILE),1)
     $(warning ***** finding all +k8s:deepcopy-gen tags)
 endif
 DEEPCOPY_DIRS := $(shell                               \
-    grep -l '+k8s:deepcopy-gen=' $(ALL_K8S_TAG_FILES)  \
+    grep --color=never -l '+k8s:deepcopy-gen=' $(ALL_K8S_TAG_FILES)  \
         | xargs -n1 dirname                            \
         | sort -u                                      \
 )
@@ -242,7 +242,7 @@ $(META_DIR)/$(DEEPCOPY_GEN).mk:
 	 INDIRECT=$$(go list -e                                \
 	     -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}'          \
 	     ./cmd/libs/go2idl/deepcopy-gen                    \
-	     | grep "^$(PRJ_SRC_PATH)"                         \
+	     | grep --color=never "^$(PRJ_SRC_PATH)"           \
 	     | sed 's|^$(PRJ_SRC_PATH)|./|'                    \
 	     | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
 	 echo $$DIRECT $$INDIRECT                              \
@@ -299,7 +299,7 @@ ifeq ($(DBG_MAKEFILE),1)
     $(warning ***** finding all +k8s:conversion-gen tags)
 endif
 CONVERSION_DIRS := $(shell                                \
-    grep '^// *+k8s:conversion-gen=' $(ALL_K8S_TAG_FILES) \
+    grep --color=never '^// *+k8s:conversion-gen=' $(ALL_K8S_TAG_FILES) \
         | cut -f1 -d:                                     \
         | xargs -n1 dirname                               \
         | sort -u                                         \
@@ -342,7 +342,7 @@ $(foreach dir, $(CONVERSION_DIRS), $(eval           \
 # but we only need to rebuild targets if the contents actually changed.  That
 # is what the .stamp file represents.
 $(foreach dir, $(CONVERSION_DIRS), $(META_DIR)/$(dir)/$(CONVERSIONS_META)):
-	TAGS=$$(grep -h '^// *+k8s:conversion-gen=' $</*.go   \
+	TAGS=$$(grep --color=never -h '^// *+k8s:conversion-gen=' $</*.go  \
 	    | cut -f2- -d=                                    \
 	    | sed 's|$(PRJ_SRC_PATH)/||');                    \
 	mkdir -p $(@D);                                       \
@@ -421,7 +421,7 @@ $(META_DIR)/$(CONVERSION_GEN).mk:
 	 INDIRECT=$$(go list -e                                \
 	     -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}'          \
 	     ./cmd/libs/go2idl/conversion-gen                  \
-	     | grep "^$(PRJ_SRC_PATH)"                         \
+	     | grep --color=never "^$(PRJ_SRC_PATH)"           \
 	     | sed 's|^$(PRJ_SRC_PATH)|./|'                    \
 	     | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
 	 echo $$DIRECT $$INDIRECT                              \

